docs/programming/using_bobs.md

```diff
@@ -0,0 +1,149 @@
+
+// Process some BOBs (save backgrounds) while doing other calculations
+bobProcessNext();
+
+// Signal no more BOBs will be added this frame
+bobPushingDone();
+
+// Process remaining BOBs (draw them)
+while(bobProcessNext()) {}
+// Or simply call bobEnd() to finish all processing
+bobEnd();
+```
+
+3. **Cleanup** (in gamestate destroy):
+```c
+bobManagerDestroy();
+```
+
+## Important BOB Functions
+
+### Initialization and Setup
+- `bobManagerCreate()`: Initializes the BOB system with front/back buffers
+- `bobInit()`: Sets up a BOB with its dimensions, position, and frame data
+- `bobReallocateBuffers()`: Allocates memory for background saving
+
+### Animation and Dimensions
+- `bobSetFrame()`: Changes a BOB's animation frame
+- `bobSetWidth()`: Modifies a BOB's width (be careful with memory allocation)
+- `bobSetHeight()`: Modifies a BOB's height (be careful with memory allocation)
+- `bobCalcFrameAddress()`: Helps calculate the memory address of a specific animation frame
+
+### Drawing Cycle
+- `bobBegin()`: Starts a new drawing cycle, undraws previous BOBs
+- `bobPush()`: Adds a BOB to the drawing queue
+- `bobProcessNext()`: Processes the next BOB operation (background save or drawing)
+- `bobPushingDone()`: Signals no more BOBs will be added this frame
+- `bobEnd()`: Completes all drawing operations
+- `bobDiscardUndraw()`: Skips undrawing (useful for first frame)
+
+## Tips and Best Practices
+
+1. **Memory Management**:
+   - For BOBs with background preservation (isUndrawRequired=1), the initial width/height should be set to the maximum anticipated size.
+   - Changing width/height to larger than initial size can cause memory corruption.
+
+2. **Performance Optimization**:
+   - Interleave blitter operations with CPU work by calling `bobProcessNext()` periodically.
+   - Don't use the blitter for other operations between `bobBegin()` and `bobEnd()`.
+   - For non-scrolling games, consider disabling Y-wrapping for better performance.
+
+3. **Animation**:
+   - Store animation frames sequentially in memory for easy frame changes.
+   - Use `bobCalcFrameAddress()` to find the correct memory address for each frame.
+
+4. **Double Buffering**:
+   - The BOB system is designed to work with double-buffered displays.
+   - You can use a single buffer by passing the same pointer for front and back buffers.
+
+5. **First Frame Handling**:
+   - Call `bobDiscardUndraw()` on the first frame to avoid undrawing BOBs that weren't previously drawn.
+
+## The BOB Structure
+
+```c
+typedef struct tBob {
+    UBYTE *pFrameData;      // Current frame bitmap data
+    UBYTE *pMaskData;       // Transparency mask data
+    tUwCoordYX pOldPositions[2]; // Previous positions for undrawing
+    tUwCoordYX sPos;        // Current position (x,y)
+    UWORD uwWidth;          // BOB width
+    UWORD uwHeight;         // BOB height
+    UBYTE isUndrawRequired; // If 1, background is preserved and restored
+    // Private fields follow...
+} tBob;
+```
```

Comment:
I wouldn't show the bob struct here since it'll change across commits and noone will update it here.

The sPos change is true, but I'll move it to same chapter as bobSetFrame() and call it "Animating BOBs" or something